uiServices/src/services.go

```diff
@@ -104,34 +105,45 @@ func getConfigs(rw http.ResponseWriter, req *http.Request){
         configPathDir = configPathDir + "/"
     }
 
-    file, err := os.Open(configPathDir + configName + ".xml")
-
+    file, err := os.Open(fmt.Sprintf("%s%s.xml", configPathDir, configName))
     if len(configPathDir) <= 1{
         rw.WriteHeader(http.StatusBadRequest)
-        logrus.Println("No Header Path Found")
+        logrus.Error("No Header Path Found")
         return
     }
-
     if err != nil {
-        logrus.Println("Configuration Name Not Found: "+configPathDir + configName)
+        logrus.Error("Configuration Name Not Found: "+configPathDir + configName)
         rw.WriteHeader(http.StatusNotFound)
         return
     }
-
     defer file.Close()
 
+    if err != nil {
+        logrus.Error("Configuration Name Not Found: "+configPathDir + configName)
+        rw.WriteHeader(http.StatusNotFound)
+        return
+    }
+
     var config perfTestUtils.Config
 
     byteValue, err := ioutil.ReadAll(file)
+    if err != nil{
+        rw.WriteHeader(http.StatusInternalServerError)
+        logrus.Error("Cannot Read File")
+        return
+    }
+
     xml.Unmarshal(byteValue, &config)
```

Comment:
`err =  xml.Unmarshal(byteValue, &config)`

otherwise you're checking for a completely different error here.